components/sidebar/sidebar_category_list/index.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {getOrderedChannelIds} from 'mattermost-redux/selectors/entities/channels';
+import {getSidebarPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {GenericAction} from 'mattermost-redux/types/actions';
+
+import SidebarCategoryList from './sidebar_category_list';
+
+// TODO: temp typing until we fix redux
+function mapStateToProps(state: GlobalState & {views: any}) {
+    // TODO: temp
+    const sidebarPrefs = getSidebarPreferences(state);
+    const lastUnreadChannel = state.views.channel.keepChannelIdAsUnread;
+    const orderedChannelIds = getOrderedChannelIds(
+        state,
+        lastUnreadChannel,
+        sidebarPrefs.grouping,
+        sidebarPrefs.sorting,
+        sidebarPrefs.unreads_at_top === 'true',
+        sidebarPrefs.favorite_at_top === 'true',
+    );
+
+    const categories = orderedChannelIds.map((channelCategory) => {
```

Comment:
You can't use `.map` in `mapStateToProps` unless it's in a selector. It returns a new array each time and that will cause `SidebarCategoryList` to re-render constantly.

Reply:
I'll leave it for now cause again this is just coming out and isn't causing any catastrophic issues currently

Reply:
Can you fix this? I'd like to start with the components re-rendering correctly since the performance of this selector is something I'm concerned about with the new sidebar.